pil-analyzer/src/type_unifier.rs

```diff
@@ -116,6 +119,13 @@ impl Unifier {
                     .zip(args2)
                     .try_for_each(|(a1, a2)| self.unify_types(a1, a2))
             }
+            (Type::Array(ArrayType { length: None, .. }), Type::Tuple(a2))
```

Comment:
I think we don't need this, at least it should not be handled in the unifier.

Reply:
Yes, I know, it was just a little hack to better understand what was going on :+1: